prometheus/testutil/promlint/validations/gauge_validations.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validations
+
+import (
+	"errors"
+	"strings"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// LintGauge detects issues specific to gauges, as well as patterns that should
+// only be used with gauges.
+func LintGauge(mf *dto.MetricFamily) []error {
```

Comment:
Hm, so this feels bit wrong. Name suggests we lint gauges, but looks like we simply lint non-gauges. I would move this to related types one by one.... but is there one?

For counters we don't need this, because _total check would be failed. For histograms and summaries, I think they are always cumulative so this check should be there too (I don't think it's implemented?).

So what we are left with? On what case you got hit by this?